serverless.yml

```diff
@@ -1,148 +1,167 @@
-              ExpirationInDays: 1
-
-    Messages:
-      Type: AWS::SQS::Queue
-      Properties:
-        QueueName: ${self:custom.sqs}
-        MessageRetentionPeriod: 1209600
-        VisibilityTimeout: 60
-        RedrivePolicy:
-          deadLetterTargetArn:
-            Fn::GetAtt:
-            - MessagesDeadLetterQueue
-            - Arn
-          maxReceiveCount: 10
-
-    MessagesDeadLetterQueue:
-      Type: AWS::SQS::Queue
-      Properties:
-        QueueName: ${self:custom.sqs}-dead-letter-queue
-        MessageRetentionPeriod: 1209600
-
-    Config:
-      Type: AWS::DynamoDB::Table
-      Properties:
-        TableName: ${self:custom.config}
-        AttributeDefinitions:
-          - AttributeName: key
-            AttributeType: S
-        KeySchema:
-          - AttributeName: key
-            KeyType: HASH
-        ProvisionedThroughput:
-          ReadCapacityUnits: 5
-          WriteCapacityUnits: 5
-
-package:
-  exclude:
-    - docs/**
-    - helpers/**
-    - test/**
-
-functions:
-
-  accept:
-    handler: functions/accept.handler
-    name: ${self:custom.prefix}-accept
-    environment:
-      CONFIG_TABLE: ${self:custom.config}
-      QUEUE_NAME: ${self:custom.sqs}
-      CONTENT_BUCKET: ${self:custom.contentBucket}
-    events:
-      - http:
-          path: accept
-          method: get
-
-  pollQueue:
-    timeout: 60
-    handler: functions/pollQueue.handler
-    name: ${self:custom.prefix}-pollQueue
-    environment:
-      CONFIG_TABLE: ${self:custom.config}
-      QUEUE_NAME: ${self:custom.sqs}
-      CONTENT_BUCKET: ${self:custom.contentBucket}
-      PROCESS_QUEUE_FUNCTION: ${self:custom.process}
-    events:
-      - schedule: rate(1 minute)
-  
-  processQueueItem:
-    timeout: 60
-    handler: functions/processQueueItem.handler
-    name: ${self:custom.process}
-    environment:
-      CONFIG_TABLE: ${self:custom.config}
-      QUEUE_NAME: ${self:custom.sqs}
-      CONTENT_BUCKET: ${self:custom.contentBucket}
\ No newline at end of file
+service: watchdog-proxy
+
+plugins:
+  - serverless-s3-remover
+  - serverless-apigw-binary
+
+custom:
+  region: ${self:provider.region}
+  stage: ${opt:stage, self:provider.stage}
+  prefix: ${self:service}-${self:custom.stage}
+  process: ${self:custom.prefix}-processQueueItem
+  config: ${self:custom.prefix}-config
+  credentials: ${self:custom.prefix}-credentials
+  sqs: ${self:custom.prefix}-messages
+  contentBucket: ${self:custom.prefix}-content
+  fnEnv:
+      NODE_ENV: ${env:NODE_ENV,"production"}
+      ENABLE_DEV_AUTH: ${env:ENABLE_DEV_AUTH,"0"}
+      SERVICE_STAGE: ${self:custom.stage}
+      SERVICE_PREFIX: ${self:custom.prefix}
+      CONFIG_TABLE: ${self:custom.config}
+      CREDENTIALS_TABLE: ${self:custom.credentials}
+      QUEUE_NAME: ${self:custom.sqs}
+      CONTENT_BUCKET: ${self:custom.contentBucket}
+      PROCESS_QUEUE_FUNCTION: ${self:custom.process}
+  remover:
+    buckets:
+      - ${self:custom.contentBucket}
+  apigwBinary:
```

Comment:
This took an annoying number of hours to learn about - useful to know for future projects that might need binary uploads. (i.e. they don't work without this enabled)